icu4j/main/tests/core/src/com/ibm/icu/dev/test/lang/UCharacterCaseTest.java

```diff
@@ -837,7 +838,12 @@ else if (lowerTest.charAt(i)
     }
 
     private void assertGreekUpper(String s, String expected) {
-        assertEquals("toUpper/Greek(" + s + ')', expected, UCharacter.toUpperCase(GREEK_LOCALE_, s));
+        Normalizer2 nfc = Normalizer2.getNFCInstance();
```

Comment:
Great stuff.

If we want to get a more complete workout, we might refactor as follows:

1. Add assertGreekUpperTitle(String s, String expectedUpper, String expectedTitle), which calls assertGreekUpper(s, expectedUpper, s)
4. Add the code to test titlecase also. 
6. When it fails, you can use the actual title values to expand the test cases, checking the actual for validity.
8. I think that titlecase is also expected to roundtrip, so you could add that also.

But this goes beyond the original ticket, so you might just clone it off into a separate ticket for the future. So approving in case you want to do that.

Reply:
If I recall correctly, there is no Greek tailoring for titlecase (that is, accents are retained in titlecase and only removed in ALL CAPS), hence the lack of dedicated tests for that.

Reply:
Yup, now that I think about it. Never mind then!